### DISCUSSION TITLE
Auto migration is not working in NestJS

### DISCUSSION THREAD

**Discussion by kevindavee** [mikro-orm/mikro-orm#4041] [Q&A] [answered]
**Describe the bug**
I would like to run the migrations when my app is starting. It works in the test, when I run my NestJS app using `nest start`, the migration is not running. Turns out that `nest start` run the code from the `dist` folder. But when I check `dist/src/migrations`, there is no migrations file.

`app.module.ts`
```typescript
@Module({
  imports: [
    MikroOrmModule.forRoot(defineConfig),
    ...modules
  ],
})
export class AppModule implements NestModule, OnModuleInit {
  constructor(private readonly orm: MikroORM) {}

  async onModuleInit(): Promise<void> {
    await this.orm.getMigrator().up();
  }

  configure(consumer: MiddlewareConsumer) {
    consumer.apply(MikroOrmMiddleware).forRoutes('*');
  }
}
```

`mikro-orm.config.ts`
```typescript
export default defineConfig({
  host: config.postgresHost,
  port: config.postgresPort,
  user: config.postgresUser,
  password: config.postgresPassword,
  dbName: config.postgresDb,
  metadataProvider: TsMorphMetadataProvider,
  entities: ['./dist/**/*.entity.js', './dist/*.entity.js'],
  entitiesTs: ['./src/**/*.entity.ts', './src/*.entity.ts'],
  debug: config.nodeEnv === 'development',
});

```

**To Reproduce**
1. Have some migrations
2. Run `nest start` or `nest start --watch`

**Expected behavior**
I expect that when I run `nest start`, the migrations should be executed.

**Versions**

| Dependency | Version |
| - | - |
| node | 18 |
| typescript | 4.3.5 |
| mikro-orm | 5.6.2 |
| your-driver | postgres |

**Comment by B4nan** [comment, marked as answer]
You probably have TS configured wrongly, imho you should not have a `disr/src` folder, it should be just `src` and `dist`, those two shouldn't be nested inside each other.

If that is what you actually want, you need to configure the ORM so it understands it, as migrations are by default checked in `src` or `dist` folders respectively, but not in a `disr/src` folder.

https://mikro-orm.io/docs/migrations#running-migrations-in-production
